tracer/src/Datadog.Trace.Tools.dd_dotnet/Checks/ProcessBasicCheck.cs

```diff
@@ -0,0 +1,463 @@
+                            foundKey = true;
+                        }
+                    }
+                }
+
+                ok &= !foundKey;
+
+                return ok;
+            }
+            catch (Exception ex)
+            {
+                Utils.WriteError(ErrorCheckingRegistry(ex.Message));
+                return true;
+            }
+        }
+
+        private static bool CheckProfilerPath(ProcessInfo process, string key, bool requiredOnLinux)
+        {
+            bool ok = true;
+
+            if (process.EnvironmentVariables.TryGetValue(key, out var profilerPath))
+            {
+                if (!IsExpectedProfilerFile(profilerPath))
+                {
+                    Utils.WriteError(WrongProfilerEnvironment(key, profilerPath));
+                    ok = false;
+                }
+
+                if (!File.Exists(profilerPath))
+                {
+                    Utils.WriteError(MissingProfilerEnvironment(key, profilerPath));
+                    ok = false;
+                }
+            }
+            else if (requiredOnLinux)
+            {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    Utils.WriteError(EnvironmentVariableNotSet(key));
+                    ok = false;
+                }
+            }
+
+            return ok;
+        }
+
+        private static bool CheckClsid(IRegistryService registry, string registryKey)
+        {
+            var profilerPath = registry.GetLocalMachineValue(registryKey);
+
+            if (profilerPath == null)
+            {
+                Utils.WriteWarning(MissingRegistryKey(registryKey));
+                return false;
+            }
+
+            if (!IsExpectedProfilerFile(profilerPath))
+            {
+                Utils.WriteError(WrongProfilerRegistry(registryKey, profilerPath));
+                return false;
+            }
+
+            if (!File.Exists(profilerPath))
+            {
+                Utils.WriteError(MissingProfilerRegistry(registryKey, profilerPath));
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool IsExpectedProfilerFile(string fullPath)
+        {
+            var fileName = Path.GetFileName(fullPath);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return "Datadog.Trace.ClrProfiler.Native.dll".Equals(fileName, StringComparison.OrdinalIgnoreCase) ||
+                       "Datadog.Tracer.Native.dll".Equals(fileName, StringComparison.OrdinalIgnoreCase) ||
+                       // Check for legacy names
+                       "Datadog.AutoInstrumentation.NativeLoader.x64.dll".Equals(fileName, StringComparison.OrdinalIgnoreCase) ||
+                       "Datadog.AutoInstrumentation.NativeLoader.x86.dll".Equals(fileName, StringComparison.OrdinalIgnoreCase);
```

Comment:
I forgot those 😢 .. I would have liked them to disappear once and for all

Reply:
Well at some point we should decide they are old enough that we should stop looking for them